egs/wsj/s5/utils/split_scp.pl

```diff
@@ -132,94 +160,129 @@
-            print STDERR "Error: split_scp.pl producing empty .scp file $scpfn (too many splits and too few speakers?)\n";
+    for (my $scpidx = 0; $scpidx < $numscps; $scpidx++) {
+
+        my $scpfn = $OUTPUTS[$scpidx];
+        open(my $fh, ">", $scpfn)
+            || die "Could not open scp file $scpfn for writing.";
+
+        my $count = 0;
+        if (@{ $scparray[$scpidx] } == 0) {
+            print STDERR
+                "Error: split_scp.pl producing empty .scp file $scpfn (too many splits and too few speakers?)\n";
             $error = 1;
         } else {
-            foreach $spk ( @{$scparray[$scpidx]} ) {
-                print F @{$spk_data{$spk}};
+            foreach my $spk (@{ $scparray[$scpidx] }) {
+                print $fh @{ $spk_data{$spk} };
                 $count += $spk_count{$spk};
             }
-            if($count != $scpcount[$scpidx]) { die "Count mismatch [code error]"; }
+
+            if ($count != $scpcount[$scpidx]) {
+                die "Count mismatch [code error]";
+            }
         }
-        close(F);
+
+        close($fh);
     }
 } else {
-   # This block is the "normal" case where there is no --utt2spk
-   # option and we just break into equal size chunks.
 
-    open(I, "<$inscp") || die "Opening input scp file $inscp";
+    # This block is the "normal" case where there is no --utt2spk
+    # option and we just break into equal size chunks.
+
+    open(my $in_fh, "<", $inscp) || die "Opening input scp file $inscp";
 
-    $numscps = @OUTPUTS;  # size of array.
-    @F = ();
-    while(<I>) {
+    my $numscps = @OUTPUTS; # size of array.
+    my @F       = ();
+    while (<$in_fh>) {
         push @F, $_;
     }
-    $numlines = @F;
-    if($numlines == 0) {
+    close $in_fh;
+    my $numlines = @F;
+    if ($numlines == 0) {
         print STDERR "split_scp.pl: error: empty input scp file $inscp , ";
         $error = 1;
     }
-    $linesperscp = int( $numlines / $numscps); # the "whole part"..
-    $linesperscp >= 1 || die "You are splitting into too many pieces! [reduce \$nj]";
-    $remainder = $numlines - ($linesperscp * $numscps);
-    ($remainder >= 0 && $remainder < $numlines) || die "bad remainder $remainder";
+
+    my $linesperscp = int($numlines / $numscps); # the "whole part"..
+    $linesperscp >= 1
+        || die "You are splitting into too many pieces! [reduce \$nj]";
+
+    my $remainder = $numlines - ($linesperscp * $numscps);
+    ($remainder >= 0 && $remainder < $numlines)
+        || die "bad remainder $remainder";
+
     # [just doing int() rounds down].
-    $n = 0;
-    for($scpidx = 0; $scpidx < @OUTPUTS; $scpidx++) {
-        $scpfile = $OUTPUTS[$scpidx];
-        open(O, ">$scpfile") || die "Opening output scp file $scpfile";
-        for($k = 0; $k < $linesperscp + ($scpidx < $remainder ? 1 : 0); $k++) {
-            print O $F[$n++];
+    my $n = 0;
+    for (my $scpidx = 0; $scpidx < @OUTPUTS; $scpidx++) {
+        my $scpfile = $OUTPUTS[$scpidx];
```

Comment:
There seems to be quite a lot of reformatting done.  This makes it very hard for me to see what the actual changes are.  This is a script that we need to be careful about changing the behavior of.  The diff is way too large.